S7.Net/PLC.cs

```diff
@@ -540,14 +540,19 @@ public void ReadClass(object sourceClass, int db, int startByteAdr = 0)
         /// <param name="startByteAdr">Start byte address. If you want to write DB1.DBW200, this is 200.</param>
         /// <param name="value">Bytes to write. If more than 200, multiple requests will be made.</param>
         /// <returns>NoError if it was successful, or the error is specified</returns>
-        public ErrorCode WriteBytes(DataType dataType, int db, int startByteAdr, byte[] value)
+        public ErrorCode WriteBytes(DataType dataType, int db, bool bitMode, int startByteAdr, int bitAdr, byte[] value)
         {
             int localIndex = 0;
             int count = value.Length;
+            if (bitMode && value.Length != 1)

```

Comment:
With separate functions, this check ins not more needed, because value becomes a boolean.